### PR TITLE
[2.21.0] Add tests to ensure empty result on query condition for invalid enum

### PR DESCRIPTION
**Pull request by nguyenv**
https://github.com/TileDB-Inc/TileDB-Py/issues/1880

This is now fixed in core. This PR just adds unit tests to ensure query conditions return an empty result, instead of erroring out, for invalid enumeration values.

### CODE REVIEW

**Comment by nguyenv** [comment]
This is also needs to be ported over to TileDB-SOMA-Py.

**Review by ihnorton — CHANGES_REQUESTED** [review]
I don't think this is the right way to solve this issue. Let's see if we can address this in libtiledb.

**Comment by nguyenv** [comment]
[sc-43002]

After discussing with @kounelisagis, we are just going to use this pre-existing PR to add the unit tests instead of opening a new one.

**Comment by kounelisagis** [comment]
Error comes from libtiledb. Should we address it there?

`FAILED tiledb/tests/test_query_condition.py::QueryConditionTest::test_qc_enumeration - tiledb.cc.TileDBError: TileDB internal: Enumeration value not found for field 'attr2'`

**Comment by nguyenv** [comment]
We need to wait for the libtiledb 2.21.0 release for this fix and then we can update `setup.py`. Although in the meantime we can test this against 2.21.0-rc2 to make sure CI is all passing.

File: setup.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 # - this is for builds-from-source
 # - release builds are controlled by `misc/azure-release.yml`
 # - this should be set to the current core release, not `dev`
-TILEDB_VERSION = "2.20.1"
+TILEDB_VERSION = "2.21.0-rc1"
```
Comment (nguyenv) [inline setup.py:23]: Reminder to revert before merging.

**Comment by nguyenv** [comment]
@kounelisagis I'm so sorry, when I merged main into this branch I accidentally removed your changes. Can you push your changes again.

**Review by ihnorton — COMMENTED** [review]
LGTM but please double-check that these all _fail_ without 2.21.

**Comment by kounelisagis** [comment]
It fails without 2.21 (or at least with 2.20...)

---

```
git checkout viviannguyen/sc-38443/dataframe-value-filter-on-enum-dict-column
pip install -r requirements_dev.txt
python setup.py develop
python -c "import tiledb; print(tiledb.libtiledb.version())"
```
**(2, 21, 0)**
```
pytest -o log_cli=true --log-cli-level=10 tiledb/tests/test_query_condition.py
```
Passes

---

```
git checkout dev setup.py
python setup.py develop
python -c "import tiledb; print(tiledb.libtiledb.version())"
```
**(2, 20, 1)**

```
pytest -o log_cli=true --log-cli-level=10 tiledb/tests/test_query_condition.py
```
Gives the following error:
https://gist.github.com/kounelisagis/7c9d8803b24f92335b2c71d58bc7f9c8

**Comment by kounelisagis** [comment]
Also tested it with 2.21.0 (not pre-release) and it works.

**Review by kounelisagis — APPROVED** [review]

**Comment by nguyenv** [comment]
Thanks for doing the work of double checking @kounelisagis.

**Comment by nguyenv** [comment]
And also for reverting the libtiledb version in setup.py.
